### PR TITLE
WIP for --insecure-registry fix

### PR DESCRIPTION
**Pull request by gigawhitlocks**
WIP because I'm handing this off to @matthewavery for the weekend (thanks dude)

What works:
Pulling from an insecure registry specified with --insecure-registry works
Pulling from an insecure registry w/o --insecure-registry fails correctly
Pulling from a secure registry works

What doesn't quite work:
Pulling an empty or nonexistent image from an insecure registry doesn't give a certificate error and instead reports that the image wasn't found (or a successful pull) -- this shouldn't happen and there should be an SSL error instead. @lcastellano and I believe that this originates from the `LearnRegistryURL` method below doing something it should not.

Also there are no tests.

### CODE REVIEW

**Comment by gigawhitlocks** [comment]
Looks like there's also a conflict that needs to be resolved :(

File: lib/imagec/docker.go [hunk 1]
```diff
@@ -47,7 +46,7 @@ const (
 	DigestSHA256EmptyTar = string(dlayer.DigestSHA256EmptyTar)
 )
 
-// FSLayer is a container struct for BlobSums defined in an image manifest
```
Comment (gigawhitlocks) [inline lib/imagec/docker.go]: and this typo snuck in somehow

File: lib/imagec/imagec_test.go [hunk 2]
```diff
@@ -1,4 +1,3 @@
-// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
```
Comment (gigawhitlocks) [inline lib/imagec/imagec_test.go]: also didnt mean to delete this

File: lib/imagec/docker.go [hunk 3]
```diff
@@ -1,4 +1,3 @@
-// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
```
Comment (gigawhitlocks) [inline lib/imagec/docker.go]: also didnt mean to delete this

**Comment by matthewavery** [comment]
I want to clarify the issue briefly. If the registry is truly ` insecure` then imagec and the portlayer behave perfectly. It is only the case of self signed certs that should report an SSL error from the insecure registry option.

**Comment by matthewavery** [comment]
After further inspection of this behavior I believe it is a non issue. I will be opening a separate PR that does a little spring cleaning and rectifies the typos that @gigawhitlocks has mentioned above. Then I will rebase. There are some merge conflicts, but it is a formating conflict that I will also resolve/prune. All changes in the new PR will be the same changes here the @lcastellano and @gigawhitlocks have made.

**Comment by matthewavery** [comment]
closing this PR. it is the parent PR to this PR(#4396). #4396 will the be the merged changes.

**Comment by gigawhitlocks** [comment]
Code was merged in #4396

**Pull request closed without merging**